src/utils.rs

```diff
@@ -20,9 +19,9 @@ where
     Ok((s[..pos].parse()?, s[pos + 1..].parse()?))
 }
 
-pub fn create_progress_bar(total: usize, message: &str) -> Arc<Mutex<ProgressBar<Stdout>>> {
```

Comment:
We could have another function, say, `create_progress_bar_arc_mutex` (i couldn't come up with a better name :) and do `Arc::new(Mutex::new(create_progress_bar(total, message))))` inside so you don't have to do it three times below.

Reply:
Done, I added a `create_syncable_progress_bar(...)` function